compiler/passes/src/processing_interpret/expression.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2019-2025 Provable Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::ProcessingInterpretVisitor;
+
+use leo_ast::{CallExpression, Expression, ExpressionReconstructor, Location, Variant};
+use leo_errors::TypeCheckerError;
+
+impl ExpressionReconstructor for ProcessingInterpretVisitor<'_> {
+    type AdditionalOutput = ();
+
+    fn reconstruct_call(&mut self, input: CallExpression) -> (Expression, Self::AdditionalOutput) {
```

Comment:
We should definitely be dis-allowing this but I think you've just set us up for FFI in the future 🚀